src/lib/support/SerializationUtils.cpp

```diff
@@ -877,18 +877,29 @@ WEAVE_ERROR ReadDataForType(TLVReader &aReader, void *aStructureData, const Fiel
 
         case SerializedFieldTypeUTF8String:
         {
-            char *dst = NULL;
             // TLV Strings are not null terminated
             uint32_t length = aReader.GetLength() + 1;
 
-            dst = (char *)memMgmt->mem_alloc(length);
-            VerifyOrExit(dst != NULL, err = WEAVE_ERROR_NO_MEMORY);
+            if (length > 1)
+            {
+                char *dst = NULL;
 
-            err = aReader.GetString(dst, length);
-            SuccessOrExit(err);
+                dst = (char *)memMgmt->mem_alloc(length);
+                VerifyOrExit(dst != NULL, err = WEAVE_ERROR_NO_MEMORY);
+
+                err = aReader.GetString(dst, length);
+                SuccessOrExit(err);
+
+                LogReadWrite("%s utf8string '%s' allocating %d bytes at %p", "R", dst, length, dst);
+
+                *static_cast<char**>(aStructureData) = dst;
+            }
+
+            else
+            {
+                *static_cast<char**>(aStructureData) = "";
```

Comment:
This seems likely to cause problems with existing code that frees the deserialized string.  Specifically, any code attempting to free the string would need to be modified to detect zero length strings and not to call mem_free() on them.  Otherwise, a crash or memory heap corruption could result.

Reply:
That's right.
I updated the patch so now there is no attempt to call mem_free() on empty strings.